test/e2e/storage/persistent_volumes.go

```diff
@@ -308,45 +310,105 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
+				ssTester := framework.NewStatefulSetTester(c)
 
-				By("Creating PVCs")
+				By("Creating a StatefulSet pod to initialize data")
+				writeCmd := "true"
 				for i := 0; i < numVols; i++ {
-					pvc = framework.MakePersistentVolumeClaim(framework.PersistentVolumeClaimConfig{}, ns)
-					pvc, err = framework.CreatePVC(c, ns, pvc)
-					Expect(err).NotTo(HaveOccurred())
-					pvcs = append(pvcs, pvc)
+					writeCmd += fmt.Sprintf("&& touch %v", getVolumeFile(i))
 				}
-
-				By("Waiting for PVCs to be bound")
-				for _, pvc := range pvcs {
-					framework.Logf("Created PVC %q", pvc.Name)
-					framework.ExpectNoError(framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, c, ns, pvc.Name, framework.Poll, framework.ClaimProvisionTimeout))
+				writeCmd += "&& sleep 10000"
+
+				probe := &v1.Probe{
+					Handler: v1.Handler{
+						Exec: &v1.ExecAction{
+							// Check that the last file got created
+							Command: []string{"test", "-f", getVolumeFile(numVols - 1)},
+						},
+					},
+					InitialDelaySeconds: 1,
+					PeriodSeconds:       1,
 				}
 
-				By("Creating a pod and initializing data")
-				writeCmd := "true"
-				for i, pvc := range pvcs {
-					// mountPath is /mnt/volume<i+1>
-					writeCmd += fmt.Sprintf("&& touch /mnt/volume%v/%v", i+1, pvc.Name)
-				}
-				pod := framework.MakePod(ns, nil, pvcs, false, writeCmd)
-				pod, err = c.CoreV1().Pods(ns).Create(pod)
+				spec := makeStatefulSetWithPVCs(ns, writeCmd, numVols, probe)
+				ss, err := c.AppsV1().StatefulSets(ns).Create(spec)
 				Expect(err).NotTo(HaveOccurred())
-				framework.ExpectNoError(framework.WaitForPodSuccessInNamespace(c, pod.Name, ns))
+				ssTester.WaitForRunningAndReady(1, ss)
 
-				By("Recreating the pod and validating the data")
-				framework.ExpectNoError(framework.DeletePodWithWait(f, c, pod))
+				By("Deleting the StatefulSet but not the volumes")
+				ss, err = ssTester.Scale(ss, 0)
+				Expect(err).NotTo(HaveOccurred())
+				ssTester.WaitForStatusReplicas(ss, 0)
+				err = c.AppsV1().StatefulSets(ns).Delete(ss.Name, &metav1.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Creating a new Statefulset and validating the data")
 				validateCmd := "true"
-				for i, pvc := range pvcs {
-					// mountPath is /mnt/volume<i+1>
-					validateCmd += fmt.Sprintf("&& test -f /mnt/volume%v/%v", i+1, pvc.Name)
+				for i := 0; i < numVols; i++ {
+					validateCmd += fmt.Sprintf("&& test -f %v", getVolumeFile(i))
 				}
-				pod = framework.MakePod(ns, nil, pvcs, false, validateCmd)
-				pod, err = c.CoreV1().Pods(ns).Create(pod)
+				validateCmd += "&& sleep 10000"
+
+				spec = makeStatefulSetWithPVCs(ns, validateCmd, numVols, probe)
+				ss, err = c.AppsV1().StatefulSets(ns).Create(spec)
 				Expect(err).NotTo(HaveOccurred())
-				framework.ExpectNoError(framework.WaitForPodSuccessInNamespace(c, pod.Name, ns))
+				ssTester.WaitForRunningAndReady(1, ss)
+
+				By("Deleting the pod and revalidating the data")
```

Comment:
Is it necessary to double validate here? We're already deleting the statefulset and bringing a new one up.